### ISSUE TITLE
SSL not supported due to arbitrary hardcodings

### ISSUE THREAD

**Issue by pfalcon** [amnong/easywebdav#1]
Subj says it all, and should give a hint for solution - if URL is going to be constructed from user input, then just let the user pass the URL instead:

```
-    def __init__(self, host, port=80, username=None, password=None):
-        self.baseurl = 'http://{}:{}'.format(host ,port)
+    def __init__(self, baseurl, username=None, password=None):
+        self.baseurl = baseurl
```

**Comment by amnong** [comment]
Hey Mr. Falcon,
Does this [https://github.com/amnong/easywebdav/pull/4] also solve this problem for you? I believe it does. Please let me know.

**Comment by brki** [comment]
I can say that #4 does not solve everything.  For example if the given URL is webdav.example.com/some/path/ , after #4 this becomes webdav.example.com/some/path/:80, which is obviously not correct.

**Comment by amnong** [comment]
Resolved as part of #18 .
